File: Python/Diverse_Filer_Fra_Tidligere_Semestere/forel_160126.py
from numpy import *
import matplotlib.pyplot as plt

# Summere tall

print 3 + 4

# Potens

print 2**4

# Divisjon

print 3/4.

# $\sin(x)$ og tallet $\pi$

print sin(pi/2)

# $e^x$

print exp(2)


# Kvadratrot


print sqrt(16)


# Regne ut logaritmer og verifisere regnereglene for disse


print log(exp(2))


# Sinus i grader
# Sinus regnes ut i radianer. Folgende regner ut i grader i stedet.


print sin(radians(pi/2))


# Presisjon
# Bruke 16 siffers presisjon i stedet


print '%.16f' % sin(radians(pi/2))


# Bruke 4 siffers presisjon


print '%.4f' % sin(radians(pi/2))


# Definere variable


a = 1
b = -2
c = a*b
print c


# Funksjoner for avrunding


print floor(2.5)
print ceil(2.5)
print round(2.5)


# Generere tilfeldige tall


print random.random(3)


# Definere matriser


A = array([[2, -3, 1], [0, 1, 3], [-4, 2, 1]])
print A



B = array([[2, 3, -1], [0, 1, -3], [4, -2, 1]])
print B


# Enkle matriseoperasjoner


print (A + B)



print (A - B)



print 3*A



print matrix(A)**7


# Komponentvis potens


print matrix(A)**7


# Transponere en matrise


print A



print A.T


# Invertere en matrise


print linalg.inv(A)


# Determinant


print linalg.det(A)


# Multiplisere matriser


print dot(A, B)


# Radvektorer og soylevektorer


b = array([-2, 5, 6])
c = array([7, -3, 2])



# Skalarprodukt og vektorprodukt


print dot(b, c)
print cross(b, c)


# Diverse maater aa lage nye matriser paa


print random.random((2, 3))



print zeros((2, 3))



print ones((2, 3))



print eye(3)


# Komponentvise operasjoner


print A*B



print A/B



print sin(A)



print exp(A*(B**2))


# Regne ut maksimum i matriser


print max([1, 2, 3])



print max([[1, 2, 3], [4, 5, 6]])



print max([[1, 7, 3], [4, 5, 6]])


# Generere lister og regne ut sum


a = arange(1, 11)
print a



print sum(a)



print sum(a**2)



a = arange(1,11,2)
print a


# Enkel plotting


t = linspace(0, 2, 100)
plt.plot(t,t**2)
plt.show()



# Kombinere matriser


print A



print B



C = hstack([A, B])
print C



C=vstack([A, B])
print C


# Trekke ut deler av en matrise


print C[0,1]



print C[0,:]



print C[:,0]



print C[0:3,1]



print C[ix_([0,2],[0,2])]

File: Matlab/Diverse_Filer_Fra_Tidligere_Semestere/forel_160126.m
%% Summere tall
3+4

%% Potens

2^4

%% Divisjon

3/4

%% $\sin(x)$ og tallet $\pi$

sin(pi/2)

%% $e^x$

exp(2)

%% Kvadratrot

sqrt(16)

%% Regne ut logaritmer og verifisere regnereglene for disse

log(exp(2))

%% Sinus i grader
% Sinus regnes ut i radianer. F?lgende funksjon regner ut i grader i stedet.

sind(pi/2)

%% Presisjon
% Bruke 16 siffers presisjon i stedet

format long
sind(pi/2)

% Tilbake til 4 siffers presisjon

format short
sind(pi/2)

%% Definere variable. 
% semikolon gj?r at ingenting blir skrevet ut

a=1
a=1;
b=-2;
c=a*b

%% Funksjoner for avrunding 

floor(2.5)
ceil(2.5)
round(2.5)

%% Generere tilfeldige tall

rand(3)

%% Definere matriser

A=[2 -3 1; 0 1 3; -4 2 1]
B=[2 3 -1; 0 1 -3; 4 -2 1];

%% Enkle matriseoperasjoner
A+B
A-B
3*A
A^7

%% Komponentvis potens
A.^7

%% Transponere en matrise
A
A'

%% Invertere en matrise

inv(A)

%% Determinant

det(A)

%% Multiplisere matriser

A*B

%% Radvektorer og s?ylevektorer

b=[-2 5 6]
c=[7; -3; 2]


%% Skalarprodukt og vektorprodukt
dot(b, c)
cross(b, c)

%% Diverse m?ter ? lage nye matriser p?.

rand(2,3)
zeros(2,3)
ones(2, 3)
eye(3)

%% Komponentvise operasjoner

A.*B
A./B
A/B
sin(A)
exp(A.*(B.^2))

%% Regne ut maksimum i matriser

max([1 2 3])
max([1 2 3; 4 5 6])
max([1 7 3; 4 5 6])

%% Generere lister og regne ut sum
a=1:10
sum(a)
sum(a.^2)
a=1:2:10
t=linspace(0,2,100)

%% Enkel plotting
plot(t,t.^2)


%% Kombinere matriser
A
B
C=[A B]
C=[A; B]

%% Trekke ut deler av en matrise
C(1,2)
C(1,:)
C(:,1)
C(1:3,2)
C([1,3],[1,3])